src/simcore_service_deployment_agent/git_url_watcher.py

```diff
@@ -18,19 +19,29 @@
 MAX_TIME_TO_WAIT_S = 10
 
 
+@retry(
+    stop=stop_after_attempt(NUMBER_OF_ATTEMPS),
+    wait=wait_fixed(1) + wait_random(0, MAX_TIME_TO_WAIT_S),
+)
 async def _git_clone_repo(
     repository: URL,
     directory: Path,
     branch: str,
     username: str = None,
     password: str = None,
 ):
-    cmd = f"git clone -n {URL(repository).with_user(username).with_password(password)} --depth 1 {directory} --single-branch --branch {branch}"
+    if username != None and password != None and username != "" and password != "":
+        # Black code linter crashed(!) here and the pre-commit hook failed. This three-liner is thus a dirty hack so I can commit in the first place... :(
+        cmd = "git clone -n "
+        cmd += str(URL(repository).with_user(username).with_password(password))
+        cmd += f" --depth 1 {directory} --single-branch --branch {branch}"
```

Comment:
I think you should also be able to pass commands using a list such as
```suggestion
        cmd = ["git clone", "-n", str(URL(repository).with_user(username).with_password(password)), "--depth 1", f"{directory}", "--single-branch", f"--branch {branch}"]
```
of course this needs adaptations on the cmd_utils.py file as well

Reply:
done